openedx/core/djangoapps/theming/apps.py

```diff
@@ -0,0 +1,83 @@
+
+import os
+import six
+from django.apps import AppConfig
+from django.conf import settings
+from django.core.checks import Error, Tags, register
+
+
+class ThemingConfig(AppConfig):
+    name = 'openedx.core.djangoapps.theming'
+    verbose_name = "Theming"
+
+
+@register(Tags.compatibility)
+def check_comprehensive_theme_settings(app_configs, **kwargs):
+    """
+    Checks the comprehensive theming theme directory settings.
+
+    Raises compatibility Errors upon:
+        - COMPREHENSIVE_THEME_DIRS is not a list
+        - theme dir path is not a string
+        - theme dir path is not an absolute path
+        - path specified in COMPREHENSIVE_THEME_DIRS does not exist
+
+    Returns:
+        List of any Errors.
+    """
+    if not getattr(settings, "ENABLE_COMPREHENSIVE_THEMING"):
+        # Only perform checks when comprehensive theming is enabled.
+        return []
+
+    errors = []
+
+    # COMPREHENSIVE_THEME_DIR is no longer supported - support has been removed.
+    if hasattr(settings, "COMPREHENSIVE_THEME_DIR"):
+        theme_dir = settings.COMPREHENSIVE_THEME_DIR
+
+        errors.append(
+            Error(
+                "COMPREHENSIVE_THEME_DIR setting has been removed in favor of COMPREHENSIVE_THEME_DIRS.",
+                hint='Transfer the COMPREHENSIVE_THEME_DIR value to COMPREHENSIVE_THEME_DIRS.',
+                obj=theme_dir,
+                id='openedx.core.djangoapps.theming.E001',
+            )
+        )
```

Comment:
BTW, it occurred to me that the only thing that'll stop this new error from invalidating most current instances is the fact that settings validation is not run on a deployed WSGI stack:
https://docs.djangoproject.com/en/1.11/topics/checks/#module-django.core.checks

I've created this sandbox to verify:
https://ct-startup.sandbox.edx.org/

which started up fine despite the following line being in both {lms|cms}.env.json:
```
    "COMPREHENSIVE_THEME_DIR": "",
```